### PR TITLE
Add support for HTTPS proxies

### PR DESCRIPTION
**Pull request by greg-poirier-swi**
The net/http package in Go supports the use of the HTTPS_PROXY environment variable to specify the URL of the proxy server. e.g. https://proxy.mydomain.com:8080. This exposes an otel.https_proxy_url value that users can use to set the environment variable.

### CODE REVIEW

File: deploy/helm/templates/metrics-deployment.yaml [hunk 1]
```diff
@@ -43,6 +43,8 @@ spec:
                   name: solarwinds-api-token
                   key: SOLARWINDS_API_TOKEN
                   optional: true
+            - name: HTTPS_PROXY
```
Comment (gantrior) [inline deploy/helm/templates/metrics-deployment.yaml:46]: I believe this is not right place to define this environment variable. It should be in https://github.com/solarwinds/swi-k8s-opentelemetry-collector/blob/master/deploy/helm/templates/common-env-config-map.yaml which is common for Otel collectors for metrics/events/logs, I believe this is applicable to all of them.

**Review by gantrior — CHANGES_REQUESTED** [review]

File: deploy/helm/templates/metrics-deployment.yaml (hunk 1 above)
Comment (greg-poirier-swi) [reply deploy/helm/templates/metrics-deployment.yaml:46]: You're right.

**Comment by greg-poirier-swi** [comment]
We should really make it so that manifest.yaml just doesn't have checksums. That entire mechanism of deploying is going away, and it's tedious and unnecessary to have to update them.

**Comment by greg-poirier-swi** [comment]
I'll squash merge and clean up the commit message. Where is the :shame: emoji.

**Review by gantrior — DISMISSED** [review]

**Review by etichy — DISMISSED** [review]

**Review by gantrior — APPROVED** [review]

**Pull request merged**